### PR TITLE
refactor: mv visible-on-scroll directive to package

### PR DESCRIPTION
**Pull request by rwd**
And register on each of the components that need it, not globally.

### CODE REVIEW

**Review by lbiedinger — COMMENTED** [review]
the `v-` prefix is applied to the directive name, which is being imported as `visibleOnScrollDirective` shouldn't the new package just be named `visibleOnScroll` or `visibleOnScrollDirective` then?

**Comment by rwd** [comment]
@lbiedinger fair point. It's been renamed to `@europeana/vue-visible-on-scroll`

File: packages/vue-visible-on-scroll/tests/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        describe('when scrolling down', () => {
+          describe('when the element is visible already (should always be the case)', () => {
+            it('updates the scrollPosition', () => {
+              const wrapper = factory();
+              const visibleOnScrollElement = wrapper.findComponent('p');
+              visibleOnScrollElement.element.scrollPosition = 20;
+              visibleOnScrollElement.element.scrolledVisible = true;
+
+              window.scrollY = 140;
+              window.dispatchEvent(new CustomEvent('scroll', {}));
```
Comment (rwd) [inline packages/vue-visible-on-scroll/tests/index.spec.js:62]: Why are the CustomEvent instantiations called with an empty object as second argument throughout these tests?

Comment (lbiedinger) [reply packages/vue-visible-on-scroll/tests/index.spec.js]: It was taken from an example that had some arguments in the custom event and then just copy pasted for all the specs, but it's not needed.
Has been removed.

**Review by lbiedinger — APPROVED** [review]

**Pull request merged**